R/ntfy.R

```diff
@@ -18,39 +18,76 @@ ntfy_server <- function(var = "NTFY_SERVER") {
   Sys.getenv(var)
 }
 
-#' Create HTTP headers for basic authentication
-#' 
-#' @param var_username environment variable in which the ntfy username is stored
-#' @param var_password environment variable in which the ntfy password is stored
-#' 
-#' @return an object with class `request` that is passed to [httr::POST()]
+#' Get the ntfy username
+#'
+#' @param var environment variable in which the username is stored
+#'
+#' @return the username with access to the protected ntfy topic
 #' @export
-#' 
-#' @examples
-#' \dontrun{
-#' Sys.setenv("NTFY_USERNAME" = "example")
-#' Sys.setenv("NTFY_PASSWORD" = "super-secret-password")
-#' 
-#' # This generates the correct header:
-#' ntfy_authorization()
-#' #> <request>
-#' #> Headers:
-#' #>   * Authorization: Basic ZXhhbXBsZTpzdXBlci1zZWNyZXQtcGFzc3dvcmQ
-#' }
-#' 
-ntfy_authorization <- function(var_username = "NTFY_USERNAME", var_password = "NTFY_PASSWORD") {
-  username = Sys.getenv(var_username)
-  password = Sys.getenv(var_password)
-  
-  if (username == "") {
-    httr::add_headers()
+ntfy_username <- function(var = "NTFY_USERNAME") {
+  Sys.getenv(var)
+}
+
+#' Get the ntfy password
+#'
+#' @param var environment variable in which the password is stored
+#'
+#' @return the password for the username with access to the protected ntfy topic
+#' @export
+ntfy_password <- function(var = "NTFY_PASSWORD") {
+  Sys.getenv(var)
+}
+
+#' Get the ntfy authorization indicator
+#'
+#' @param var environment variable in which the ntfy authorization indicator is stored
+#'
+#' @return a logical that indicates if password authorization is used
+#' @export
+ntfy_auth <- function(var = "NTFY_AUTH") {
+  if (Sys.getenv(var) == "TRUE") {
+    TRUE
```

Comment:
(can't make the suggestion over deleted lines, so don't accept this change)
```suggestion
ntfy_auth <- function(var = "NTFY_AUTH") {
  toupper(Sys.getenv(var)) == "TRUE"
```
An `if/else` returning booleans is on my watch-list from reviewing PRs full-time - you can just return the result of the comparison. 

To be slightly more robust, uppercase the result in case someone sets `'true'`.

Reply:
Done!